### DISCUSSION TITLE
Partition coefficient (water/container) cannot be found

### DISCUSSION THREAD

**Discussion by tztz117** [Open-Systems-Pharmacology/Forum#949] [Questions & Problems]
I am adding a spatical structrure of breast using kidney as the container template. Followed Dallmann 2018 paper and when trying to run simulation this error message showed, could someone help to take a look?

![path cant find](https://user-images.githubusercontent.com/86744277/125855883-41dd99e6-6dbf-4268-8ea6-b3b9762bbe18.jpg)
[Clarithromycin lactation.zip](https://github.com/Open-Systems-Pharmacology/MoBi/files/6826176/Clarithromycin.lactation.zip)

**Comment by tztz117** [comment]
![path cant find](https://user-images.githubusercontent.com/86744277/128246990-023aee87-37f5-485c-acb8-545ba1c23ae2.jpg)
 Anyone help to take a look at this? thanks very much!

**Comment by msevestre** [comment]
You will need to add this parameter specifically in the spatial structure. It is probably outlined in the paper.
If not, search the forum for the section "how do define a tumor model" or sthg like this.
Let me know if this helps

**Comment by tztz117** [comment]
thank you very much, Michael! I have followed the steps in that tumor model and the paper of adding spatial structure, 

The interstitial/water partition was added under neighborhood as below:

![image](https://user-images.githubusercontent.com/86744277/128367924-d637006f-59cc-4e79-894b-dfbed325be2e.png)

I found this path under reactions, but can't find it in the spatial structure:

![issue path error 2](https://user-images.githubusercontent.com/86744277/128368813-6e719b49-30d0-467c-9790-706883ebd148.JPG)

**Comment by tztz117** [comment]
in front of this path, it has "..|", when I trying to add one by myself for testing, it don't have "..|", what it means?

**Comment by msevestre** [comment]
..| is a way to specify where to find the parameter. So it will be in the parent container in this case
".." can be read as "parent"
Maybe check some of the documentation 
https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/first-steps

**Comment by tztz117** [comment]
Hi Michael,

I checked discussion "Define partition coefficient (int/water) when adding a new compartment Open-Systems-Pharmacology/MoBi#302". I have added partition cofficient (interstitial/water) as guided by the paper and those discussions. but the error message here is "partition coefficient (water/container) can not be found. how to find this? I can't find it in other tissues either.



![path cant find](https://user-images.githubusercontent.com/86744277/128422836-89406ad2-93a7-4ace-a9f1-49969536d515.jpg)

**Comment by msevestre** [comment]
Can you attach your project? I'll check it out tomorrow

**Comment by tztz117** [comment]
[Clarithromycin lactation.zip](https://github.com/Open-Systems-Pharmacology/MoBi/files/6941839/Clarithromycin.lactation.zip)

Yes, Michael. Thank you very much!

I am also wondering whehter it has something to do with the model, which has both parent and metabolite.

**Comment by msevestre** [comment]
@tztz117 Ok I looked at your model and there are quite a few things that are wrong I believe.
I do not think that you followed the steps described in the tutorial
-  I would suggest looking at the step by step instructions again
- In your milk compartment, the parameter `Partition coefficient (water/container)` is missing (spatial structure). This is clearly what the message in MoBi is telling you. 
- In fact, the milk compartment should have the same structure as any other compartment probably. 
- What tutorial did you follow?

Last but not least. If you need help, you need to create an issue in the Forum. MoBi issues are only to report bugs or feature requests. 

I will move the issue for you

**Comment by msevestre** [comment]
@tztz117 it is important that your milk compartment follows the same structure as any other compartment defined in the model

**Reply by tztz117** [reply]
Hi Michael, thank you very much! I followed Dallman's 2018 paper of "Applied Concepts in PBPK Modeling: How to Extend
an Open Systems Pharmacology Model to the Special Population of Pregnant Women", the steps in the supplemental material. He used heart as the template, but here i used kidney as the template since I need the milk compartment in the breast, which I think may be similar to urine in kidney. I will compare the milk compartment with others to see what is missing there. thank you very much!

**Reply by AndreDlm** [reply]
Dear @tztz117, the tutorial explains the process of adding a new **organ** to a PBPK model, i.e. 4 new compartments (blood cells, plasma, interstitial and intracellular compartments of the new organ) ; in your case, I understood that you want to add only 1 compartment for the milk into an existing organ for the breast which requires another workflow. 
In short, the issue is that you have a reaction defined in the new compartment and to calculate the kinetics of this reaction the `Partition coefficient (water/container)` is needed. This is not described in the tutorial above, but there are two options to solve it:
* follow @msevestre's advice above and add the parameter in your milk compartment
* Inactivate CYP3A4 expression/reaction in the milk compartment (e.g. by excluding this compartment with tags from the reaction building block or by inactivating CYP3A4 expression in the milk compartment e.g. by unchecking `Is Present` in the molecule start values building block)

**Comment by tztz117** [comment]
Dear Andre and Michael, thank you very much for helping address the issue! I will try and let you know.  

I like the flexibility of Mobi but I am wondering whether there is workshop or training to teach how to use it at a relatively adavanced level like adding a tissue compartment, I find several similar discussions about those errors.
